test/python/pulse/test_pulse_lib.py

```diff
@@ -31,7 +31,7 @@ def test_sample_pulse(self):
         n_samples = 100
         samples = np.linspace(0, 1., n_samples, dtype=np.complex128)
         name = 'test'
-        sample_pulse = SamplePulse(samples, name=name)
```

Comment:
Did you create the `TestWaveform` class? We should keep all the previous tests with a deprecation notice, and also test the new `Waveform` with the same tests (copied)